src/bscPlugin/validation/BrsFileValidator.ts

```diff
@@ -82,12 +84,45 @@ export class BrsFileValidator {
             NamespaceStatement: (node) => {
                 this.validateDeclarationLocations(node, 'namespace', () => util.createBoundingRange(node.keyword, node.nameExpression));
 
-                node.parent.getSymbolTable().addSymbol(
-                    node.name.split('.')[0],
-                    node.nameExpression.range,
-                    node.getType(),
-                    SymbolTypeFlags.typetime
-                );
+                // Since namespaces are accessed via a DottedGetExpression, we need to build the types for
+                // the top level namespace, all the way down.
```

Comment:
This code is very ugly --- but there were lots of edge cases, depending on the order various namespaces (and sub namespaces) were declared in.